jupyterlab_caip_optimizer/src/store/studies.ts

```diff
@@ -0,0 +1,68 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { Study, AsyncState } from '../types';
+import { OptimizerService } from '../service/optimizer';
+import { ServerProxyTransportService } from 'gcp_jupyterlab_shared';
+
+const optimizer = new OptimizerService(new ServerProxyTransportService());
+
+export const fetchStudies = createAsyncThunk<Study[]>(
+  'studies/fetch',
+  async () => {
+    // TODO: use metadata service instead of hardcoded values
```

Comment:
Just want to note that we are hardcoding in project id and region data since we were unable to get the metadata server working.

Reply:
I think in that case I would still add functions (e.g. getProjectId and getRegion) to optimizer.ts and return the constants there. That way, when you add in the functionality you only need to edit the functions